src/components/VerticalNavigation/VerticalNavigationItem.js

```diff
@@ -0,0 +1,215 @@
+      nextDepth === 'secondary' ? secondaryCollapsed : tertiaryCollapsed;
+    const collapseNextNav =
+      nextDepth === 'secondary'
+        ? this.collapseSecondaryNav
+        : this.collapseTertiaryNav;
+
+    return (
+      <ListGroupItem
+        listItem // Renders as <li>. Other props can change this, see logic in react-bootstrap's ListGroupItem.
+        className={cx({
+          [`${nextDepth}-nav-item-pf`]:
+            depth !== 'tertiary' && children && children.length > 0,
+          active: trackActiveState, // This is the only class we have at the tertiary depth.
+          'visible-xs-block': trackActiveState, // TODO is trackActiveState the right thing here, and is it named right?
+          'is-hover': depth !== 'tertiary' && trackHoverState,
+          // This class is present at primary and secondary depths if mobileItem is true,
+          // except for the primary depth, where it is only present if showMobileSecondary is also true.
+          'mobile-nav-item-pf':
+            mobileItem &&
+            ((depth === 'primary' && showMobileSecondary) ||
+              depth === 'secondary'),
+          // This class is confusingly named, but the logic is more readable.
+          'mobile-secondary-item-pf':
+            mobileItem && depth === 'primary' && showMobileTertiary,
+          // I don't know, that's just how this stuff was in patternfly-ng...
+        })}
+        onMouseEnter={() => {
+          handleHover[depth]('TODO args here');
+        }}
+        onMouseLeave={() => {
+          handleBlur[depth]('TODO args here');
+        }}
+      >
+        <a onClick={() => { handleClick[depth]('TODO args here') }}> {/* TODO handleClick takes all 3 items? figure that out... */}
+          {depth === 'primary' &&
+            iconStyleClass && (
+              <OverlayTrigger
+                placement="bottom"
+                overlay={navCollapsed ? <Tooltip>{title}</Tooltip> : null}
+              >
+                <span
+                  className={cx(iconStyleClass, { hidden: hiddenIcons })}
+                  title={title}
+                />
+              </OverlayTrigger>
+            )}
+          <span className="list-group-item-value">{title}</span>
+          {this.renderBadges(badges)}
+        </a>
+        {children &&
+          children.length > 0 && (
+            <div className="nav-pf-secondary-nav"> {/* TODO should this class sometimes say tertiary? */}
+              <div className="nav-item-pf-header">
+                <a
+                  className={cx(`${nextDepth}-collapse-toggle-pf`, {
+                    collapsed: nextDepthCollapsed,
+                  })}
+                  onClick={() => {
+                    collapseNextNav('TODO args here');
+                  }} // TODO what is the item arg in these calls?  some subset of props? pass what we need from it instead.
+                />
+                <span>{title}</span>
+              </div>
+              <ListGroup>
```

Comment:
@priley86 , here is the cause of the <li>-inside-<li> warnings we are getting. This ListGroup sits inside a ListGroupItem, and renders ListGroupItem children.

Reply:
I think you need to add `componentClass="ul"` here.